python/activator/middleware_interface.py

```diff
@@ -328,12 +359,17 @@ def run_pipeline(self, visit: Visit, snaps: set) -> None:
             in the `visit` object, but we'll have to test how that works once
             we implemented this with actual data.
         """
+        # TODO: can we move this from_pipeline call to prep_butler?
         where = f"detector={visit.detector} and exposure in ({','.join(str(x) for x in snaps)})"
-        pipeline = SimplePipelineExecutor.from_pipeline(self.pipeline, where=where, butler=self.butler)
+        executor = SimplePipelineExecutor.from_pipeline(self.pipeline, where=where, butler=self.butler)
+        # TODO DM-34202: hack around a middleware bug.
+        executor.butler = Butler(butler=self.butler,
+                                 collections=[self.output_collection],
+                                 run=self.output_run)
         _log.info(f"Running pipeline {self.pipeline} on visit '{visit}', snaps {snaps}")
```

Comment:
This is a regression of the previous issue where it would put a detailed pipeline dump in the log. I think we're better off having no log message at all.

Reply:
This now has a much better log message.